### ISSUE TITLE
LinkButton: Classes are not correctly rendered when either no text or no icon is set

### ISSUE THREAD

**Issue by chovyy** [primefaces/primefaces#7753]
**Describe the defect**
LinkButton styles are not correctly rendered when either no `value` or no `icon` is given. Addtional classes `ui-button ui-widget ui-state-default ui-corner-all ui-button-text-icon-right` are added to the main `span` overriding the `ui-button-text-only` or `ui-button-icon-only` class.

This issue was introduced in PrimeFaces 10. In PrimeFaces 8 it worked fine and LinkButton behaved like CommandButton for example.

**Reproducer**
https://github.com/chovyy/primefaces-test-linkbutton-icon-style
But also visible in:
http://www.primefaces.org/showcase/ui/button/linkButton.xhtml
Compare with:
https://www.primefaces.org/showcase-v8/ui/button/linkButton.xhtml

**Environment:**
 - PF Version: _10.0_

**To Reproduce**
1. Add LinkButton without Icon to your page:
   ```xml
   <p:linkButton value="No Icon" href="#" />
   ```
   will be rendered as:
   ```html
   <span id="..." class="ui-linkbutton ui-button ui-widget ui-state-default ui-corner-all ui-button-text-only ui-button ui-widget ui-state-default ui-corner-all ui-button-text-icon-right" role="button">
     <a href="#">
       <span class="ui-button-text ui-c">No Icon</span>
     </a>
   </span>
   ```
2. Add LinkButton without Text to your page:
   ```xml
   <p:linkButton icon="pi pi-thumbs-down" href="#" />
   ```
   will be rendered as:
   ```html
   <span id="..." class="ui-linkbutton ui-button ui-widget ui-state-default ui-corner-all ui-button-icon-only ui-button ui-widget ui-state-default ui-corner-all ui-button-text-icon-right" role="button">
     <a href="#">
       <span class="ui-button-icon-left ui-icon ui-c pi pi-thumbs-down"></span>
       <span class="ui-button-text ui-c">ui-button</span>
     </a>
   </span>
   ```

**Expected behavior**
Both examples should be rendered with less classes:

1.
   ```html
   <span id="..." class="ui-linkbutton ui-button ui-widget ui-state-default ui-corner-all ui-button-text-only" role="button">
     <a href="#">
       <span class="ui-button-text ui-c">No Icon</span>
     </a>
   </span>
   ```
2.
   ```html
   <span id="..." class="ui-linkbutton ui-button ui-widget ui-state-default ui-corner-all ui-button-icon-only" role="button">
     <a href="#">
       <span class="ui-button-icon-left ui-icon ui-c pi pi-thumbs-down"></span>
       <span class="ui-button-text ui-c">ui-button</span>
     </a>
   </span>
   ```

**Comment by melloware** [comment]
Is this a duplicate of this already fixed issue: https://github.com/primefaces/primefaces/issues/7245  ??

Have you tried 11.0.0-SNAPSHOT?

**Comment by chovyy** [comment]
> Is this a duplicate of this already fixed issue: #7245 ??

Well, looks like it. I'm sorry.

> Have you tried 11.0.0-SNAPSHOT?

No, while the issue template says "PrimeFaces test can easily be set to use the latest `master-SNAPSHOT` of what is in GitHub.", there is actually no hint on how to do that.

**Comment by tandraschko** [comment]
just check the readme here in the primefaces repo but its currently a bit broken: https://github.com/primefaces/primefaces/issues/7739
